### DISCUSSION TITLE
Detect if current workspace is empty then automatically move to workspace Z

### DISCUSSION THREAD

**Discussion by mvaneijgen** [nikitabobko/AeroSpace#1013] [questions-and-answers]
I would love to check when a window closes if the current workspace is empty and then switch to a workspace of my choosing 

**Steps**
1. Doing something in workspace X 
2. Closes current window in workspace X, which is the only window left in this workspace
3. Workspace is now empty 
4. Automatically move to workspace Z

## Checklist

- [x] I've searched for duplicates. My search queries were:
  - detect empty workspace 
  - empty workspace 
- [x] I've read the guide https://nikitabobko.github.io/AeroSpace/guide
- [x] I've searched in the documentation (https://nikitabobko.github.io/AeroSpace/guide & https://nikitabobko.github.io/AeroSpace/commands). My search queries were:
  - empty
- Also [Search for configs by other users on GitHub](https://github.com/search?q=path%3A*aerospace.toml&type=code) for inspiration 
  -  https://github.com/search?q=detect+empty+path%3A*aerospace.toml&type=code

I have the feeling this should be possible with the knowledge in the docs, but with the info I've gathered I can't see how to combine everything in to a function that does this. I would love some sort of "cookbook" in the docs with examples and descriptions how to do certain things, that allows me personally to easier adept things to my own needs. Happily volunteer to write some things out!

```
aerospace CLI client version: 0.14.2-Beta 0cb8dbdfc5ee73b8cbc200f175f467ebead55201
AeroSpace.app server version: 0.14.2-Beta 0cb8dbdfc5ee73b8cbc200f175f467ebead55201
```

**Comment by hwamil** [comment]
Couldn't you bind a key to close all windows then move to z?

Or simply close all windows and move to z yourself? I don't think something like this needs a builtin feature.

Edit:

You could implement something like this with exec-and-forget

```
#!/bin/bash

if [ -z "$(aerospace list-windows --focused)" ]; then
  aerospace workspace z
fi
```
